engine-tests/src/test/java/org/terasology/metatesting/MTEClientSystemTest.java

```diff
@@ -0,0 +1,84 @@
+package org.terasology.metatesting;
+
+import org.junit.jupiter.api.Test;
+import org.terasology.engine.context.Context;
+import org.terasology.engine.core.ComponentSystemManager;
+import org.terasology.engine.core.GameEngine;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.engine.integrationenvironment.ModuleTestingHelper;
+import org.terasology.engine.integrationenvironment.jupiter.IntegrationEnvironment;
+import org.terasology.engine.network.NetworkMode;
+import org.terasology.engine.registry.In;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@IntegrationEnvironment(networkMode = NetworkMode.LISTEN_SERVER)
+public class MTEClientSystemTest {
+
+    @In
+    private ModuleTestingHelper helper;
+
+    @Test
+    public void testClientSystemExecution() throws java.io.IOException {
+        System.out.println("Starting testClientSystemExecution...");
+
+        // Create a client
+        Context clientContext = helper.createClient();
+        System.out.println("Client Context created.");
+
+        // Set a safety timeout for the run loop
+        helper.setSafetyTimeoutMs(30000);
+
+        // Inspect ComponentSystemManager
+        ComponentSystemManager csm = clientContext.get(ComponentSystemManager.class);
+        System.out.println("Client CSM: " + csm);
+        System.out.println("Client CSM Active? " + (csm != null ? csm.isActive() : "null"));
+
+        // Register a test system on the client
+        TestSystem testSystem = new TestSystem();
+        if (csm != null) {
```

Comment:
This is a unit test, so it is acceptable and actually expected to fail with null values here.